### ISSUE TITLE
[Bug]: data source grafana_users global scope error

### ISSUE THREAD

**Issue by N3X14** [grafana/terraform-provider-grafana#1734]
### Terraform Version

v1.9.3

### Terraform Grafana Provider Version

3.6.0

### Grafana Version

Grafana v11.2.0-73830 (8d98c9eba7)

### Affected Resource(s)

data.grafana_users

### Terraform Configuration Files

```hcl

provider "grafana" {
  alias = "serviceaccount"

  url  = grafana_cloud_stack.main.url
  auth = grafana_cloud_stack_service_account_token.cloud_sa.key
}

data "grafana_users" "all_users" {
  provider = grafana.serviceaccount
}
```

### Expected Behavior

No error should be returned and the requested data should be made available for consumption.

### Actual Behavior

Error: global scope resources cannot be managed with an API key. Use basic auth instead
 
   with data.grafana_users.all_users,
   on data.tf line 21, in data "grafana_users" "all_users":
   21: data "grafana_users" "all_users" {

### Steps to Reproduce

1. Create a new project
2. Create a service_account based provider configuration
3. Add the data source
4. terraform plan/apply to get the error

### Important Factoids

Using Grafana Cloud. Creating a user in grafana.net and using basicAuth errors in 401 suggesting that's not working/available either.
This is an all new project, there's no interfering state.

**Comment by julienduchesne** [comment]
Can you describe your intended use of the provider? Users are not queryable in Grafana Cloud right now because they are provisioned externally

**Comment by N3X14** [comment]
Sure thing. I'm trying to provision the whole stack (folders, folder_permissions, etc.) based on the users email addresses. We're using OAuth for login and basically all I need would be to provide an email and get the user_id in return. That was my intended use-case for this datasource. 

Not sure if I missed what you're describing in the documentation, but I'm pretty certain that was not mentioned there?

Thanks!

**Comment by N3X14** [comment]
In particular here: [Grafana Cloud Docs](https://grafana.com/docs/grafana-cloud/developer-resources/api-reference/http-api/user/)
It mentions that it would be available using basicAuth? 
Hella confusing with the various authentication options.

**Comment by julienduchesne** [comment]
> It mentions that it would be available using basicAuth?

Yeah, basicAuth would be `user:password` which isn't available in Grafana Cloud. I will take a look if it's possible to search users with a service account token. Weird that it has never come up before 🤔

**Comment by N3X14** [comment]
Yeap, tried that as well (user:password). Given it was referenced in the documentation I suspected they have some sync from grafana.net to the individual stacks to make that work. Kinda odd that basicAuth is mentioned quite a lot in Grafana Cloud docs if it's not even available in the first place x.x

**Comment by cfarrugia** [comment]
Hit the same exact problem today. I've tried using a cloud_access_policy_token in the provider (i use this successfully to create things like stacks themselves), and then the error changed to:  "the Grafana client is required for this resource. Set the auth and url provider attributes"

**Comment by julienduchesne** [comment]
👍. Grafana Cloud has an API to get a stack's users. I am looking into integrating it in the provider. Probably as a new datasource: `data.grafana_cloud_stack_users`

**Comment by N3X14** [comment]
Much appreciated!

**Comment by julienduchesne** [comment]
For info, this is the API call that can be used: https://github.com/grafana/grafana-com-public-clients/blob/ac10800522a38443066abcca234272895af2c569/go/gcom/api_instances.go#L2281-L2288

**Comment by pepordev** [comment]
any plans to implement this? we are hitting too and it is quite frustrating that we can not manage folder permissions with terraform as there is no way to fetch the users properly.

**Comment by pepordev** [comment]
> For info, this is the API call that can be used: https://github.com/grafana/grafana-com-public-clients/blob/ac10800522a38443066abcca234272895af2c569/go/gcom/api_instances.go#L2281-L2288

Thanks for the info, i managed to do what i needed, just in case it is usefull for anyone else, i used the http terraform provider to query the api using the grafana cloud access policy token (stack-users list required).

```

data "http" "grafana_users" {
  url = "https://grafana.com/api/instances/{ your_instance_id}/users"

  # Optional request headers
  request_headers = {
    Accept = "application/json"
    Authorization = " Bearer { your_access_policy_token }"
  }
}
```

Then just parse the response with 
```
locals {
  grafana_users = { for user in jsondecode(data.http.grafana_users.response_body).items : user.email => user.id }
}
```
Now you should be able to assign permissions to users using the email
```
    permissions {
      user_id    = local.grafana_users["email"]
      permission = permissions.value.permission
    }
```

**Comment by Jguer** [comment]
The following route uses organization admin privileges rather than server admin privileges, so it should work for your use case. Tell me if you run into any problems testing it.

https://registry.terraform.io/providers/grafana/grafana/latest/docs/data-sources/organization_user
